internal/command/postgres/postgres.go

```diff
@@ -231,15 +232,45 @@ func leaderIpFromNomadInstances(ctx context.Context, addrs []string) (string, er
 	return "", fmt.Errorf("no instances found with leader role")
 }
 
+func isLeader(machine *api.Machine) bool {
+	return machineRole(machine) == "leader" || machineRole(machine) == "primary"
+}
+
 func pickLeader(ctx context.Context, machines []*api.Machine) (*api.Machine, error) {
 	for _, machine := range machines {
-		if machineRole(machine) == "leader" || machineRole(machine) == "primary" {
+		if isLeader(machine) {
 			return machine, nil
 		}
 	}
 	return nil, fmt.Errorf("no active leader found")
 }
 
+func pickNewLeader(ctx context.Context, app *api.AppCompact, machines []*api.Machine, primary_region string) (*api.Machine, error) {
+	for _, machine := range machines {
+		if !isLeader(machine) && machine.HealthCheckStatus().AllPassing() && machine.Region == primary_region && passesDryRun(ctx, app, machine) == nil {
+			return machine, nil
+		}
+	}
+	return nil, fmt.Errorf("no leader could be chosen. no new leader could be found that passed health checks and currently run within the primary region")
```

Comment:
If we can't find a new leader, we should see if we can provide additional context as to why.

```
$ fly status --app shaun-flex-stable
ID            	STATE  	ROLE   	REGION	CHECKS            	IMAGE                             	CREATED             	UPDATED
6e82dd03c74587	started	replica	ord   	3 total, 3 passing	flyio/postgres-flex:15.2 (v0.0.40)	2023-03-02T02:18:31Z	2023-05-11T14:07:05Z
5683dde7fd918e	started	primary	ord   	3 total, 3 passing	flyio/postgres-flex:15.2 (v0.0.40)	2023-03-02T02:19:01Z	2023-04-25T15:06:49Z
3287299b020ee8	stopped	error  	ord   	3 total           	flyio/postgres-flex:15.2 (v0.0.40)	2023-03-02T02:19:33Z	2023-05-11T14:07:55Z

$ ./bin/flyctl pg failover --app shaun-flex-stable
Performing a failover
Connecting to fdaa:0:2e26:a7b:f4:1b9b:dac8:2... complete
Error promoting new leader, restarting existing leader
Waiting for old leader to finish stopping
Clearing existing machine lease...
Trying to start old leader
Old leader started succesfully
Error: Failed to run failover: no leader could be chosen. no new leader could be found that passed health checks and currently run within the primary region
```

I suspect this is failing because quorum can't be met, however, I can't be certain without some additional information.

Reply:
Is the way I'm doing it now better? Here's an example:
```
billy@laptop2 ~/W/flyctl (pg_failover_flex)> fly status -a yuryuryur
ID            	STATE  	ROLE   	REGION	CHECKS            	IMAGE                             CREATED             	UPDATED
918577d0f14248	started	replica	ord   	3 total, 3 passing	flyio/postgres-flex:15.2 (v0.0.40)2023-05-09T01:34:54Z	2023-05-11T19:04:09Z	
3287d49a6d1185	stopped	error  	ord   	3 total           	flyio/postgres-flex:15.2 (v0.0.40)2023-05-09T01:35:11Z	2023-05-11T19:05:46Z	
148edde6c37468	started	primary	ord   	3 total, 3 passing	flyio/postgres-flex:15.2 (v0.0.40)2023-05-09T01:34:36Z	2023-05-11T18:58:57Z	

billy@laptop2 ~/W/flyctl (pg_failover_flex)> go run . pg failover -a yuryuryur
WARN The running flyctl agent (v0.0.0-1683832270+dev) is older than the current flyctl (v0.0.0-1683832366+dev).
WARN The out-of-date agent will be shut down along with existing wireguard connections. The new agent will start automatically as needed.
Performing a failover
Connecting to fdaa:1:3ad4:a7b:f3:4671:27a8:2... complete
Error promoting new leader, restarting existing leader
Waiting for old leader to finish stopping
Clearing existing machine lease...
Trying to start old leader
Old leader started succesfully
Error: Failed to run failover: no leader could be chosen. Here are the reasons why:
148edde6c37468: already leader
3287d49a6d1185: 1+ health checks are not passing
918577d0f14248: Running a dry run of `repmgr standby switchover` failed. Try running `fly ssh console -u postgres -C 'repmgr standby switchover -f /data/repmgr.conf --dry-run' -s -a yuryuryur` for more information. This was most likely due to the requirements for quorum not being met.

please fix one or more of the above issues, and try again
```